Library/Homebrew/formula_installer.rb

```diff
@@ -351,7 +351,7 @@ def install
       message = <<~EOS
         #{formula.name} #{formula.linked_version} is already installed
       EOS
-      if formula.outdated? && !formula.head?
+      if Homebrew::EnvConfig.no_install_upgrade? && formula.outdated? && !formula.head?
         message += <<~EOS
           To upgrade to #{formula.pkg_version}, run:
             brew upgrade #{formula.full_name}
```

Comment:
I think there may be other cases where this kicks in. Perhaps worth searching the codebase for `brew upgrade`. There was the one I linked you to in `cmd/install.rb`, I think?

Reply:
Correct, this is a special case handling keg-only formulae. I've fixed it now.

Reply:
I grep'd for "To upgrade to" and only found the two cases that I've now addressed. Resolving this, but let me know if you suspect I'm missing something.